### ISSUE TITLE
Support for built-in variables

### ISSUE THREAD

**Issue by GoogleCodeExporter** [rpbouman/kettle-cookbook#48]
```
Hi!

You will find a doc file where I try to explain my problem.
```

Original issue reported on code.google.com by `joseph.c...@gmail.com` on 18 Jun 2012 at 8:32

Attachments:
- [kettle-cookbook_bug_20120618.doc](https://storage.googleapis.com/google-code-attachments/kettle-cookbook/issue-48/comment-0/kettle-cookbook_bug_20120618.doc)

**Comment by joaociocca** [comment]
from what I understand about the cookbook process, it just reads the jobs and transformations XML files, it won't execute, interpret or replace user variables - and I doubt it really should, actually...

**Comment by rpbouman** [comment]
Yeah it's a tough nut. One the one hand it is useful to replace some built-in variables - for example, I usually use job dir, trans dir and so on to keep related transformations and jobs together. I'd really like that to be picked up and documented all after expanding those vars. 

But user vars are functional elements that can help make a job / transformation dynamic. If you push another value into those, you don't suddenly get a new job or trans that needs to be documented; rather, I would want to document the fact that a variable is used, not what people put into that variable.

Keeping the issue open in case someone else has thoughts on this.
